### PR TITLE
[ci] Update rust in ci image (1.75 and 2024-01-22)

### PR DESCRIPTION
**Pull request by alvicsam**
cc https://github.com/paritytech/ci_cd/issues/926

### CODE REVIEW

File: .github/workflows/fmt-check.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
         os: ["ubuntu-latest"]
     runs-on: ${{ matrix.os }}
     container:
-      image: paritytech/ci-unified:bullseye-1.74.0-2023-11-01-v20231204
```
Comment (ggwpez) [inline .github/workflows/fmt-check.yml:18]: Can you search for `2023-11-01` in the code base and replace them please?  
I know that we have the version hard-coded in some spots.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4993106

**Comment by alvicsam** [comment]
bot fmt -v CMD_IMAGE=paritytech/ci-unified:bullseye-1.75.0-2024-01-22-v20240109

File: substrate/frame/contracts/fixtures/build.rs [hunk 2]
```diff
@@ -254,7 +254,7 @@ fn invoke_riscv_build(current_dir: &Path) -> Result<()> {
 		.expect("failed to execute process");
 
 	if build_res.status.success() {
-		return Ok(())
+		return Ok(());
```
Comment (ggwpez) [inline substrate/frame/contracts/fixtures/build.rs:257]: Is this enforced by the new formatter?  
Otherwise please revert..

**Review by ggwpez — APPROVED** [review]
Thanks!

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4993912

**Review by sergejparity — APPROVED** [review]

File: substrate/frame/contracts/fixtures/build.rs (hunk 2 above)
Comment (alvicsam) [reply substrate/frame/contracts/fixtures/build.rs]: Sorry, vscode decided to add them, fixed

**Review by athei — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4995766

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4996050

**Comment by alvicsam** [comment]
bot update-ui latest --rust_version=1.75

File: bridges/snowbridge/parachain/pallets/system/src/lib.rs [hunk 3]
```diff
@@ -79,6 +79,7 @@ use xcm_executor::traits::ConvertLocation;
 #[cfg(feature = "runtime-benchmarks")]
 use frame_support::traits::OriginTrait;
 
+#[allow(unused_imports)] // export can be unused since this is a library crate.
 pub use pallet::*;
```
[the remark this answers is not in the document]
Comment (ggwpez) [reply bridges/snowbridge/parachain/pallets/system/src/lib.rs]: lol, that explains it 😆

**Comment by liamaharon** [comment]
> @ggwpez thanks for the help with clippy! It seems I need some more help with frame ui tests: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5006699

Looking into this

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -21,7 +21,8 @@ workflow:
     - if: $CI_COMMIT_BRANCH
 
 variables:
-  CI_IMAGE: !reference [.ci-unified, variables, CI_IMAGE]
+  # CI_IMAGE: !reference [.ci-unified, variables, CI_IMAGE]
```
Comment (liamaharon) [inline .gitlab-ci.yml:24]: is this needed?

**Review by liamaharon — APPROVED** [review]

File: .gitlab-ci.yml (hunk 4 above)
Comment (alvicsam) [reply .gitlab-ci.yml:24]: yes, I'll revert it back when we change the image in the common settings

**Comment by alvicsam** [comment]
Could someone from @paritytech/bridges-core approve this PR plz?

**Review by serban300 — APPROVED** [review]

**Review by svyatonik — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Comment by mordamax** [comment]
> bot update-ui latest --rust_version=1.75

@alvicsam  didn't you have to pass the image same as you did with fmt?

**Review by bkontur — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5076401

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5077764

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5079660

**Comment by rcny** [comment]
What is the preferred solution when Clippy complains about unused imports? Just remove them or add `#[allow(unused_imports)]`? I removed them, but I noticed the second approach in the code.

**Comment by bkchr** [comment]
For the tests the `unused_imports` is fine.

**Pull request merged**